### PR TITLE
bignum: constant time implementation for bn_mod()

### PR DESCRIPTION
**Pull request by romanz**

### CODE REVIEW

**Comment by prusnak** [comment]
@jhoenicke ACK?

File: test_curves.py [hunk 1]
```diff
@@ -261,6 +261,13 @@ def test_mod(curve, r):
     lib.bn_mod(y, int2bn(curve.p))
     assert bn2int(y) == x % curve.p
 
+def test_mod_specific(curve):
+    p = curve.p
+    for x in [0, 1, 2, p - 2, p - 1, p, p + 1, p + 2, 2*p - 2, 2*p - 1, 2*p]:
```
Comment (jhoenicke) [inline test_curves.py]: Is `2*p` a good test case?  I think this is forbidden by the contract.

Comment (romanz) [inline test_curves.py]: You are right - I will fix it and re-push.

**Comment by prusnak** [comment]
Thanks, merged.

**Comment by romanz** [comment]
Thanks :)

**Pull request merged**